### PR TITLE
[feedback-widget-combine] added rating support to present_feedback_wi…

### PR DESCRIPTION
**Pull request by frknbasaran**
…dget() method.

### CODE REVIEW

File: lib/countly.js [hunk 1]
```diff
@@ -2391,6 +2420,39 @@
                 document.getElementById('countly-surveys-wrapper-' + feedback._id).style.display = "block";
                 document.getElementById('csbg').style.display = "block";
             }
+
+            /**
+             * Function to prepare rating sticker and popup
+             * @param  {Object} feedback - feedback object
+             */
+            function showRating(feedback) {
```
Comment (ArtursKadikis) [inline lib/countly.js:2428]: Rename to "showRatingForFeedbackWidget"

File: lib/countly.js [hunk 2]
```diff
@@ -2391,6 +2420,39 @@
                 document.getElementById('countly-surveys-wrapper-' + feedback._id).style.display = "block";
                 document.getElementById('csbg').style.display = "block";
             }
+
+            /**
+             * Function to prepare rating sticker and popup
```
Comment (ArtursKadikis) [inline lib/countly.js:2425]: Comment should also mention feedback widget

File: lib/countly.js [hunk 3]
```diff
@@ -2299,85 +2303,110 @@
...
                 else if (presentableFeedback.type === "nps") {
-                    document.getElementById('countly-surveys-wrapper-' + presentableFeedback._id).style.display = "block";
+                    document.getElementById('countly-' + feedbackType + '-wrapper-' + presentableFeedback._id).style.display = "block";
                     document.getElementById('csbg').style.display = "block";
                 }
+                else if (presentableFeedback.type === "rating") {
+                    var ratingShown = false;
+
+                    //Set popup show policy
+                    switch (presentableFeedback.showPolicy) {
```
Comment (ArtursKadikis) [inline lib/countly.js:2389]: this switch can be simplified

File: lib/countly.js [hunk 4]
```diff
@@ -2299,85 +2303,110 @@
...
+                        case "afterPageLoad":
+                            if (document.readyState === "complete") {
```
Comment (ArtursKadikis) [inline lib/countly.js:2391]: Refactor this. Make it better

File: lib/countly.js [hunk 5]
```diff
@@ -2201,17 +2205,20 @@
                 //Origin is passed to the popup so that it passes it back in the postMessage event
 
                 var iframe = document.createElement("iframe");
-                iframe.name = "countly-surveys-iframe";
-                iframe.id = "countly-surveys-iframe";
                 iframe.src = url;
+                iframe.name = "countly-" + feedbackType + "-iframe";
+                iframe.id = "countly-" + feedbackType + "-iframe";
+
+                loadCSS(this.url + '/surveys/stylesheets/countly-surveys.css');
```
Comment (ArtursKadikis) [inline lib/countly.js:2212]: Move into the "widgetFamily" "if" statement

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]
Reapply code styling according to the SDK's lint file.

**Comment by ArtursKadikis** [comment]
@turtledreams test this in a sample app and double check that all widget types and was to use them are working. That means 4 different scenarios:
* old rating widget approach. 
* new rating widget approach with the feedback widget mechanisms. Be sure not to use any calls from the old approach.
* surveys
* nps

File: lib/countly.js [hunk 6]
```diff
@@ -2262,12 +2262,28 @@
                 else if (presentableFeedback.type === "survey") {
                     url += "/feedback/survey";
                 }
+                else if (presentableFeedback.type === "rating") {
```
Comment (turtledreams) [inline lib/countly.js:2265]: perfect

**Review by turtledreams — APPROVED** [review]

**Pull request merged**